packages/@vue/cli-plugin-e2e-nightwatch/nightwatch.config.js

```diff
@@ -1,36 +1,98 @@
 // http://nightwatchjs.org/gettingstarted#settings-file
 
+const path = require('path')
 const deepmerge = require('deepmerge')
+const chromedriver = require('chromedriver')
+const geckodriver = require('geckodriver')
+
 const userOptions = JSON.parse(process.env.VUE_NIGHTWATCH_USER_OPTIONS || '{}')
+const useSelenium = process.env.VUE_NIGHTWATCH_USE_SELENIUM === '1'
+const startHeadless = process.env.VUE_NIGHTWATCH_HEADLESS === '1'
+const concurrentMode = process.env.VUE_NIGHTWATCH_CONCURRENT === '1'
+const chromeArgs = []
+const geckoArgs = []
+
+if (startHeadless) {
+  chromeArgs.push('headless')
+  geckoArgs.push('--headless')
+}
 
-module.exports = deepmerge({
+const defaultSettings = {
   src_folders: ['tests/e2e/specs'],
   output_folder: 'tests/e2e/reports',
-  custom_assertions_path: ['tests/e2e/custom-assertions'],
-
-  selenium: {
-    start_process: true,
-    server_path: require('selenium-server').path,
-    host: '127.0.0.1',
-    port: 4444,
-    cli_args: {
-      'webdriver.chrome.driver': require('chromedriver').path
-    }
-  },
-
+  page_objects_path: 'tests/e2e/page-objects',
+  custom_assertions_path: 'tests/e2e/custom-assertions',
+  custom_commands_path: 'tests/e2e/custom-commands',
+  test_workers: concurrentMode,
   test_settings: {
     default: {
-      selenium_port: 4444,
-      selenium_host: 'localhost',
-      silent: true
+      detailed_output: !concurrentMode,
+      launch_url: '${VUE_DEV_SERVER_URL}'
     },
 
     chrome: {
       desiredCapabilities: {
         browserName: 'chrome',
-        javascriptEnabled: true,
-        acceptSslCerts: true
+        chromeOptions: {
+          w3c: false,
+          args: chromeArgs
+        }
+      }
+    },
+
+    firefox: {
+      desiredCapabilities: {
+        browserName: 'firefox',
+        alwaysMatch: {
+          acceptInsecureCerts: true,
+          'moz:firefoxOptions': {
+            args: geckoArgs
+          }
+        }
+      },
+      webdriver: useSelenium ? {} : {
+        server_path: require('geckodriver').path,
```

Comment:
If firefox is supported by default, then we need to install `geckodriver` for users (in `packages/@vue/cli-plugin-e2e-nightwatch/generator/index.js`).
Otherwise, there should be a try catch here.

Personally I think `geckodriver` could be an [optional peer dependency](https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-optional-peer-dependencies.md). As it's not that widely used and its size is not negligible (2MB-5MB depending on OS).

Reply:
I think I missed that. Regarding distributing gecko by default, I suppose that is your decision to make. I am not entirely up to date with how widely used Firefox is in the wider dev/testing community, but I believe geckodriver is the most W3C Webdriver Api compliant browser driver and also, as far as I can tell, it is being developed by the team who originally built Webdriver. Besides, Firefox is an independent browser with a strong focus on privacy and security, and I think its values align well with Vue's.

Reply:
That makes sense.
We can include this by default for now. Later when we implement the prompt interface, we can check for the user's installed browsers and ask them which webdriver they are going to use.